colour/algebra/interpolation.py

```diff
@@ -31,23 +37,598 @@
+                warning(('"x" independent variable is not uniform, '
+                         'unpredictable results may occur!'))
+
+            self._x = value
+
+            if self._window is not None:
+                self._x_p = np.pad(
+                    self._x, (self._window, self._window),
+                    'linear_ramp',
+                    end_values=(
+                        np.min(self._x) - self._window * value_interval[0],
+                        np.max(self._x) + self._window * value_interval[0]))
+
+    @property
+    def y(self):
+        """
+        Getter and setter property for the dependent and already known
+        :math:`y` variable.
+
+        Parameters
+        ----------
+        value : array_like
+            Value to set the dependent and already known :math:`y` variable
+            with.
+
+        Returns
+        -------
+        array_like
+            Dependent and already known :math:`y` variable.
+        """
+
+        return self._y
+
+    @y.setter
+    def y(self, value):
+        """
+        Setter for the **self.y** property.
+        """
+
+        if value is not None:
+            value = np.atleast_1d(value).astype(self._dtype)
+
+            assert value.ndim == 1, (
+                '"y" dependent variable must have exactly one dimension!')
+
+            self._y = value
+
+            if self._window is not None:
+                self._y_p = np.pad(self._y, **self._padding_args)
+
+    @property
+    def window(self):
+        """
+        Getter and setter property for the half window.
+
+        Parameters
+        ----------
+        value : int
+            Value to set the half window with.
+
+        Returns
+        -------
+        int
+            Half window.
+        """
+
+        return self._window
+
+    @window.setter
+    def window(self, value):
+        """
+        Setter for the **self.window** property.
+        """
+
+        if value is not None:
+            assert is_integer(value), '"window" must be an integer!'
+
+            assert value >= 1, '"window" must be equal or superior to 1!'
+
+            self._window = value
+
+            if self._x is not None:
+                self.x = self._x
+
+            if self._y is not None:
+                self.y = self.y
+
+    @property
+    def kernel(self):
+        """
+        Getter and setter property for the kernel callable.
+
+        Parameters
+        ----------
+        value : callable
+            Value to set the kernel callable.
+
+        Returns
+        -------
+        callable
+            Kernel callable.
+        """
+
+        return self._kernel
+
+    @kernel.setter
+    def kernel(self, value):
+        """
+        Setter for the **self.kernel** property.
+        """
+
+        if value is not None:
+            assert hasattr(
+                value,
+                '__call__'), ('"{0}" attribute: "{1}" is not callable!'.format(
+                    'kernel', value))
```

Comment:
Should we have an `is_kernel` function that hides this check (and maybe tests additional functionality the kernel should provide)?

Reply:
Okay, yeah, I agree this should be fine with the updated docs